cmp/grouping/GroupingChooserModel.ts

```diff
@@ -183,10 +183,21 @@ export class GroupingChooserModel extends HoistModel {
             }
         });
 
+        this.addReaction({
+            track: () => this.dimensions,
+            run: () => this.cleanStaleDims()
+        });
+
         this.setValue(value);
         this.setFavorites(favorites);
     }
 
+    @action
+    setDimensions(dimensions: Array<DimensionSpec | string>) {
+        this.dimensions = this.normalizeDimensions(dimensions);
+        this.dimensionNames = keys(this.dimensions);
+    }
+
     @action
     setValue(value: string[]) {
         if (!this.validateValue(value)) {
```

Comment:
Should we have a "louder" error handle? Maybe a toast along with the logWarn on line 204.

Reply:
We discussed showing a toast within setValue prob not a great idea - could be called rapidly / repeatedly by app reaction, etc. At the same time, we talked about how current handling is pretty subtle, with warning log only. I am not sure there is a better option here, or that we want to get into it with this change.